apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/CommonQueries.java

```diff
@@ -0,0 +1,202 @@
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.JoinType;
+import javax.persistence.criteria.Root;
+
+/** Common database queries that can be shared between V1 and V2. */
+public class CommonQueries {
+
+  /**
+   * Database function that checks all claims for a given beneficiaryId and returns a bitwise mask
+   * value that shows if a given claim type will have any data.
+   */
+  public static final String CHECK_CLAIMS_FOR_DATA_SQL =
+      "SELECT * FROM check_claims_mask(:beneIdValue)";
+
+  /**
+   * Database function that takes in a patient identifier type and value and returns a bigint
+   * BENE_ID if the identifier can find a beneficiaries record. Supported identifier types are: mbi,
+   * mbi-hash, hicn-hash.
+   */
+  public static final String FIND_BENE_ID_FROM_IDENTIFIER_SQL =
+      "SELECT * FROM find_beneficiary(:searchIdType, :searchIdValue)";
+
+  /** preclude construction from outsiders. */
+  private CommonQueries() {}
+
+  /**
+   * Query database to determine which claim types have data for the specified beneficiary. The
+   * database function returns a bitwise mask value that denotes which claim types will have data
+   * for the specified beneficiaryId. This represents a fast and efficient way to ignore a requested
+   * claim type that ultimately has no data for our beneficiaryId.
+   *
+   * <p>The database function returns bits as follows: CARRIER_CLAIMS : bit 0 INPATIENT CLAIMS : bit
+   * 1 OUTPATIENT CLAIMS : bit 2 SNF CLAIMS : bit 3 DME CLAIMS : bit 4 HHA CLAIMS : bit 5 HOSPICE
+   * CLAIMS : bit 6 PART D CLAIMS : bit 7
+   *
+   * <p>For more information on the database function, see:
+   * V111__SETUP_CLAIMS_AVAILABILITY_FUNCTION.SQL in the db migration directory.
+   *
+   * @param entityManager {@link EntityManager} used to query database.
+   * @param beneficiaryId used to identify the Beneficiary to check claims for.
+   * @return int bitmask denoting which claims have data.
+   */
+  public static int availableClaimsData(EntityManager entityManager, long beneficiaryId) {
+    List<Object> values =
+        entityManager
+            .createNativeQuery(CHECK_CLAIMS_FOR_DATA_SQL)
+            .setParameter("beneIdValue", beneficiaryId)
+            .getResultList();
+
+    return (int) (values != null && values.size() > 0 ? values.get(0) : 0);
+  }
+
+  /**
+   * Search database for a BENE_ID using a combination of search type and search value. Supports use
+   * case where a beneficiary's MBI and/or HICN may have changed by looking at both current
+   * (beneficiaries) and historical (beneficiaries_history) data for a match.
+   *
+   * <p>For more information on the database function, see:
+   * V115__FIND_BENEFICIARY_IDENTIFIER_FUNCTION.SQL in the db migration directory.
```

Comment:
This was recently updated to _V116__Find_Beneficiary_Identifier_Function.sql_, wasn't it?

Reply:
duh...don't know why I had 115 on brain...it is 116 and the sql is named correctly...thx.